runtime/src/consensus/tendermint/verifier.rs

```diff
@@ -203,6 +226,20 @@ impl Verifier {
         Ok(())
     }
 
+    fn latest_consensus_state(&self, instance: &mut Instance) -> Result<ConsensusState, Error> {
+        let verified_block = instance
+            .light_client
+            .verify_to_highest(&mut instance.state)
+            .map_err(|err| Error::VerificationFailed(err.into()))?;
+        self.update_insecure_posix_time(&verified_block);
+
+        let state_root = state_root_from_header(&verified_block.signed_header);
```

Comment:
Should we also update the cache, e.g. `verified_state_roots`?

Reply:
The `verified_state_roots` are the runtime state roots while the above are consensus layer state roots (and `latest_state` is only dealing with consensus layer state roots). The light blocks that are fetched as part of `verify_to_highest` are already cached in the `LruStore` (see initialization in the `run` method), but the problem with `latest_state` is that it doesn't know what the latest height is, so it will in any case request this information from the node on each invocation.

But given that the node is notifying the runtime of new consensus layer blocks as they are finalized (which results in invoking the `sync` method of the verifier) we can record that and avoid the extra query on each `latest_state` request. I'll change it.